BLUESPAWN-client/headers/util/permissions/permissions.h

```diff
@@ -105,12 +114,36 @@ namespace Permissions {
 		PACL GetSACL() const;
 	};
 
+	class LsaHandleWrapper : public GenericWrapper<LSA_HANDLE> {
+	public:
+		LsaHandleWrapper(LSA_HANDLE handle) :
+			GenericWrapper(handle, std::function<void(LSA_HANDLE)>(SafeCloseLsaHandle), nullptr) {};
+		static void SafeCloseLsaHandle(LSA_HANDLE handle);
+	};
+
 	/*Enum for storing type of Owner an Owner object is*/
 	enum OwnerType {
 		NONE, USER, GROUP 
 	};
 
 	class Owner : public Loggable {
+		//Policy Handler for User Rights Assignment Tracking
+		static LsaHandleWrapper lPolicyHandle;
+
+		//Boolean to track if the handle wrapper is initialized
+		static bool bPolicyInitialized;
+
+		//List of super user privileges
+		static const std::vector<std::wstring> vSuperUserPrivs;
+
+		/**
+		* Function to initialize lPolicyHandle and set bPolicyInitialized
+		* to true if initialization succeeded.
+		*/
+		static void InitializePolicy();
+
+		//Let LsaHandleWrapper handle initialization tracking
+		friend void LsaHandleWrapper::SafeCloseLsaHandle(LSA_HANDLE handle);
```

Comment:
Unless I'm misunderstanding, this is here so that SafeCloseLsaHandle can reach into this class and messing with static members. From a design perspective, I'd much prefer that LsaHandleWrapper wraps an LSA_HANDLE and is independent of Owner. It just makes more sense that way.